public/diagnostic.js

```diff
@@ -23,3 +23,22 @@
 //  -  Order 1 : 2 hats ($5 each)
 //  -  Order 2 : 1 desk lamp ($20 each)
 //  -  Order 3 : 3 hand towels ($8 each)
+
+
+import Ember from 'ember';
+
+const order = Ember.Object.extend({
+  orderPrice: function (price, quantity) {
+    return price * quantity;
+  }
+});
+
+const Cart = Ember.Object.extend({
+  orders: [],
+
+  addToCart(orders) {
+    this.get('orders').pushObject(orders);
+  }
+});
+
```

Comment:
```js

const Order = Ember.Object.extend({
  orderPrice: Ember.computed('price', 'quantity', function(){
    return this.get('price') * this.get('quantity');
  })
});

const Cart = Ember.Object.extend({
  orders: [],
  addToCart: function(order){
    this.get('orders').addObject(order);
  },
  totalPrice: Ember.computed('orders','orders.@each.orderPrice', function(){
    let orders = this.get('orders');
    return orders.map((order) => order.get('orderPrice')).reduce((m,e) => m+e);
  })
});

``` 

very close here Kosta, I included the solutions of those so you can see the subtle missteps in syntax you have. technically neither of these is complete, but you're on the right track and I appreciate inclusion of the resource you used. 

1/5 revisit this and pseudo code out the rest before trying it again or reviewing the full solution